adr/0007-choosing-an-open-source-component.md

```diff
@@ -0,0 +1,63 @@
+# Choosing an open source component
+
+Date: 2018-01-09
+
+## Status
+
+Proposed.
+
+## Context 
+
+Using poor quality or unmaintained third-party software can be at the least inconvenient, and at the most downright dangerous: for example, it can fail to take advantage of the benefits from the latest technology, and security holes can go unpatched.
+
+Using third-party software that is difficult to maintain integrated in our environment can make updating it difficult, time-consuming or risky.
+
+Using software with an inappropriate license may present a legal risk, or may risk others choosing not to use our software because of the burden the license subsequently places on them.  
+
+We want to minimise exposure to these risks when choosing an open source component to use in the eLife code base.   
+            
+## Decision
+When choosing an open source component, care should be taken to try to choose something that is:
+
+- licensed appropriately
+- reliable
+- well maintained
+- not complex to maintain in our environment
+
+### Must have
+ - compatible license
+ 
+### Nice to have
+ - issue tracker
+ - many maintainers
+ - community enthusiasm
+ - documentation
+ - low complexity to maintain in our environment
+ 
+## Warning signs
+ - many open issues
+ - customised license
+ - few commits
+ - exotic code
+ - delays in 3rd party PRs getting merged or rejected for no obvious reason
+
+During the inclusion of a library in a pull request, complete the following list and let the maintainers of the project make an informed choice.
+
+- [] license type:
+- [] is the license customised?
+- [] the there an issue tracker?
+- [] open issue count:
+- [] commit count:
+- [] maintainer count:
+- anticipated effort to maintain in our environment:
```

Comment:
If there was something quite complex, but was only used in one place in your environment that might be described as "low", and something thats simpler to manage but used almost everywhere might be described as "high". 

Could add something for how often its planned to be used?

Reply:
You're onto something about the additional directions that can influence how big the dependency is.

This (which refers to [technologies in general](https://blog.codeship.com/growing-tech-stack-say-no/), not just libraries or frameworks) came to mind:
<img src="https://1npo9l3lml0zvr6w62acc3t1-wpengine.netdna-ssl.com/wp-content/uploads/2015/11/Risks-of-introducing-a-new-tool.jpg" />